rules/group.go

```diff
@@ -84,16 +95,22 @@ type Group struct {
 // DefaultEvalIterationFunc is the default implementation.
 type GroupEvalIterationFunc func(ctx context.Context, g *Group, evalTimestamp time.Time)
 
+// ShouldEvaluateRuleGroupFunc will be used to determine if a rule group
+// needs to be evaluated or deactivated. Deactivating a rule group will
+// reset the metrics but not unload the rule group.
+type ShouldEvaluateRuleGroupFunc func(ctx context.Context, g *Group, evalTimestamp time.Time, doneCh <-chan struct{}) bool
+
 type GroupOptions struct {
-	Name, File        string
-	Interval          time.Duration
-	Limit             int
-	Rules             []Rule
-	ShouldRestore     bool
-	Opts              *ManagerOptions
-	QueryOffset       *time.Duration
-	done              chan struct{}
-	EvalIterationFunc GroupEvalIterationFunc
+	Name, File                  string
+	Interval                    time.Duration
+	Limit                       int
+	Rules                       []Rule
+	ShouldRestore               bool
+	Opts                        *ManagerOptions
+	QueryOffset                 *time.Duration
+	done                        chan struct{}
+	EvalIterationFunc           GroupEvalIterationFunc
+	ShouldEvaluateRuleGroupFunc ShouldEvaluateRuleGroupFunc
```

Comment:
I know other fields are not commented, but can we at least break this pattern and explain how this options should be used by the callers? What is the contract here? What could be surprising?